File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/module/ModuleClassLoader.java
package de.jalumu.magma.platform.base.module;

import lombok.SneakyThrows;

import java.io.File;
import java.net.URL;
import java.net.URLClassLoader;

public class ModuleClassLoader extends URLClassLoader {

    public ModuleClassLoader(ClassLoader parent) {
        super(new URL[]{}, parent);
    }

    @SneakyThrows
    public void addJar(File jar) {
        super.addURL(jar.toURI().toURL());
    }

}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/module/ModuleClassLoader.kt
package de.jalumu.magma.platform.base.module

import lombok.SneakyThrows
import java.io.File
import java.net.URLClassLoader

class ModuleClassLoader(parent: ClassLoader?) : URLClassLoader(arrayOf(), parent) {
    @SneakyThrows
    fun addJar(jar: File) {
        super.addURL(jar.toURI().toURL())
    }
}
